server/internal/infrastructure/memory/project.go

```diff
@@ -0,0 +1,137 @@
+package memory
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/reearth/reearth-cms/server/internal/usecase"
+	"github.com/reearth/reearth-cms/server/internal/usecase/repo"
+	"github.com/reearth/reearth-cms/server/pkg/id"
+	"github.com/reearth/reearth-cms/server/pkg/project"
+	"github.com/reearth/reearth-cms/server/pkg/rerror"
+)
+
+type Project struct {
+	lock sync.Mutex
+	data map[id.ProjectID]*project.Project
```

Comment:
```suggestion
	data util.SyncMap[id.ProjectID, *project.Project]
```

There is a very useful struct because it uses generics and sync.Mutex is no longer needed